upgrade-instructions.md

```diff
@@ -12,23 +12,325 @@ software documentation. Do not follow this guide blindly!
+## 0.11.0
 
-### Set up TTL on the label_change table
-Date: 15/09/2023
+- [Change default minimum ngram settings for contact indices](#change-default-minimum-ngram-settings-for-contact-indices)
 
-JIRA: https://issues.apache.org/jira/browse/JAMES-3940
+### Change default minimum ngram settings for contact indices
+Date: 01/10/2024
 
-Concerned products: Distributed James, Distributed James ESv6
+Issue: https://github.com/linagora/tmail-backend/issues/1204
 
-From now on, we set a default 60 days (configurable) Cassandra time-to-live on the `label_change` table
-to reduce considered outdated changes' storage usage.
+Optional to adapt as the effection is minor.
 
-If one want to keep the old behavior which is not using Cassandra time-to-live on those JMAP change tables, please set the
-following configurations in `cassandra.properties`:
+Minimum ngram (minimum characters to autocomplete) now defaults to 2 (previously was 3).
 
+To adapt the change, we need to migrate contact indices to a new version with the new indices settings.
+
+#### Migrate Domain contact index
+
+- Step 1: create a v2 index
+
+```curl
+PUT /domain_contact_v2
+{
+  "settings": {
+    "index": {
+      "max_ngram_diff": "27",
+      "number_of_shards": "5",
+      "number_of_replicas": "1",
+      "analysis": {
+        "filter": {
+          "ngram_filter": {
+            "type": "ngram",
+            "min_gram": "2",
+            "max_gram": "29"
+          },
+          "edge_ngram_filter": {
+            "type": "edge_ngram",
+            "min_gram": "2",
+            "max_gram": "29"
+          },
+          "preserved_ascii_folding_filter": {
+            "type": "asciifolding",
+            "preserve_original": "true"
+          }
+        },
+        "analyzer": {
+          "email_ngram_filter_analyzer": {
+            "filter": [
+              "ngram_filter",
+              "lowercase"
+            ],
+            "type": "custom",
+            "tokenizer": "uax_url_email"
+          },
+          "rebuilt_keyword": {
+            "filter": [
+              "lowercase"
+            ],
+            "type": "custom",
+            "tokenizer": "keyword"
+          },
+          "name_edge_ngram_filter_analyzer": {
+            "filter": [
+              "edge_ngram_filter",
+              "lowercase",
+              "preserved_ascii_folding_filter"
+            ],
+            "type": "custom",
+            "tokenizer": "standard"
+          }
+        }
+      }
+    }
+  },
+  "mappings": {
+    "properties": {
+      "contactId": {
+        "type": "keyword"
+      },
+      "domain": {
+        "type": "keyword"
+      },
+      "email": {
+        "type": "text",
+        "analyzer": "email_ngram_filter_analyzer",
+        "search_analyzer": "rebuilt_keyword"
+      },
+      "firstname": {
+        "type": "text",
+        "analyzer": "name_edge_ngram_filter_analyzer",
+        "search_analyzer": "standard"
```

Comment:
Is this intended to keep standard here?

Reply:
I think so. What is the expectation to not keep search_analyzer as standard?

Reply:
Having a `standard` search analyzer means filtering.

Example

GIVEN `Andre` 
WHEN I type `and` in the search bar
THEN the standard analyzer thinks it is an english stopword and filters out the token thus rendering the search useless.

Treating the search as a keyword (tokenized the right way) would avoid such pitfalls.

I'm interested by the test of andre/and BTW to see if I am right.

Reply:
Relevant documentation: https://opensearch.org/docs/latest/analyzers/

Reply:
`Keyword` seem like the analyzer we want to use here, given that we don't want to split the emails before indexing.

Reply:
> I'm interested by the test of andre/and BTW to see if I am right.

I added a test. We should be good already.

Reply:
> Overall Iam sory I was convinced the standard analyzer was doing more work here.

Don't be sorry. It is always good to check though.